src/basekernels/nn.jl

```diff
@@ -0,0 +1,24 @@
+"""
+    NeuralNetworkKernel()
+
+Neural network kernel function.
+
+```math
+    κ(x, y) =  asin(x' * y / sqrt[(1 + x' * x) * (1 + y' * y)])
+```
+# Significance:
+Neal(1996)  pursued the limits of large models, and showed that a Bayesian neural network becomes a Gaussian process with a **neural network kernel** as the number of units approaches infinity.
```

Comment:
```suggestion
Neal (1996) pursued the limits of large models, and showed that a Bayesian neural network becomes a Gaussian process with a **neural network kernel** as the number of units approaches infinity.
```
Moreover, IIRC that was for a network with one hidden layer and a specific activation function? In that case that should be noted here.